matlab/src/+mlarrow/+util/table2mlarrow.m

```diff
@@ -0,0 +1,87 @@
+function [variables, metadata] = table2mlarrow(t)
+%TABLE2MLARROW Converts a MATLAB table into a form
+%   suitable for passing to the mlarrow C++ MEX layer.
+%
+%   [VARIABLES, METADATA] = TABLE2MLARROW(T)
+%   Takes a MATLAB table T and returns struct array equivalents
+%   which are suitable for passing to the mlarrow C++ MEX layer.
+%
+%   VARIABLES is an 1xN struct array representing the the table variables.
+%
+%   VARIABLES contains the following fields:
+%
+%   Field Name     Class        Description
+%   ------------   -------      ----------------------------------------------
+%   Name           char         Variable's name
+%   Type           char         Variable's MATLAB datatype
+%   Data           numeric      Variable's data
+%   Valid          logical      0 = invalid (null), 1 = valid (non-null) value
+%
+%   METADATA is a 1x1 struct array with the following fields:
+%
+%   METADATA contains the following fields:
+%
+%   Field Name    Class         Description
+%   ------------  -------       ----------------------------------------------
+%   Description   char          Table description (T.Properties.Description)
+%   NumRows       double        Number of table rows (height(T))
+%   NumVariables  double        Number of table variables (width(T))
+%   Version       double        Feather file version to write
+%
+%   See also FEATHERREAD, FEATHERWRITE.
+
+% Licensed to the Apache Software Foundation (ASF) under one or more
+% contributor license agreements.  See the NOTICE file distributed with
+% this work for additional information regarding copyright ownership.
+% The ASF licenses this file to you under the Apache License, Version
+% 2.0 (the "License"); you may not use this file except in compliance
+% with the License.  You may obtain a copy of the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+% implied.  See the License for the specific language governing
+% permissions and limitations under the License.
+
+import mlarrow.util.*;
+
+% Currently, featherwrite only supports Version 2 Feather files.
+featherVersion = 2;
```

Comment:
I'm curious why this detail needs to be here at all

Reply:
The original intent of explicitly specifying the Feather format version here was to protect against future format changes that might break the current `featherwrite` code. As the Feather format evolves, we would eventually like to react to changes, but, we wanted to explicitly lock down the behavior of writing version 2 for the time being.

However, to your point, the Feather format version we are writing has nothing to do with converting a MATLAB `table` into an intermediate struct representation which is suitable for conversion into an `arrow::Table`/`arrow::RecordBatch` (which is the purpose of `mlarrow.util.table2mlarrow`). To separate concerns, I am in favor of moving this out of `mlarrow.util.table2arrow` and into the top-level `featherwrite.m` code. I was thinking we could pass a third `options` argument to `featherwritemex`, which would basically be a struct containing only the format version to write for now. In the future, we might want to add more options to control C++-level writing behavior.

From a design perspective, I imagine a future where a `"Version"` name-value pair could be specified like the following:

``>> featherwrite(filename, t, "Version", 3);``

This would control what Feather file version is written under the hood.

Reply:
Thinking a bit more about this, it doesn't look like there is a way to control what version of the Feather file format is being written using `arrow::ipc::feather::TableWriter`. So, for the time being, I'll go ahead and remove the use of `featherVersion` entirely. I can add it back later if it makes sense in the future.

Reply:
Fixed.